### PR TITLE
*: fix duplicated table name when join

### PR DESCRIPTION
**Pull request by siddontang**
Check duplicated join table name, following queries are all invalid. 
```
select * from t1 join t1;
select * from t1 join t2 as t1;
select * from t1 join test.t1;
```

### CODE REVIEW

File: rset/rsets/join.go [hunk 1]
```diff
@@ -161,6 +161,38 @@ func (r *JoinRset) buildPlan(ctx context.Context, node interface{}) (plan.Plan,
...
+
+	// first check ident name
+	identName := t.String()
+	_, ok := r.tableNames[identName]
+	if ok {
+		return errors.Errorf("%s: duplicate name %s", r, identName)
+	}
+	r.tableNames[identName] = true
+
+	qualifedName := tr.Schema + "." + tr.Name
```
Comment (qiuyesuifeng) [inline rset/rsets/join.go]: qualifiedName?

File: rset/rsets/join.go [hunk 2]
```diff
@@ -161,6 +161,38 @@ func (r *JoinRset) buildPlan(ctx context.Context, node interface{}) (plan.Plan,
...
+	// we should check qualifed name too, e,g select * form t1 join test.t1
+	if identName != qualifedName {
+		_, ok = r.tableNames[qualifedName]
+		if ok {
+			return errors.Errorf("%s: duplicate name %s", r, t.String())
```
Comment (qiuyesuifeng) [inline rset/rsets/join.go]: t.String() -> identName?

File: rset/rsets/join.go [hunk 3]
```diff
@@ -161,6 +161,38 @@ func (r *JoinRset) buildPlan(ctx context.Context, node interface{}) (plan.Plan,
...
+		if ok {
+			return errors.Errorf("%s: duplicate name %s", r, t.Name)
+		}
+		r.tableNames[t.Name] = true
+		return nil
+	}
+
+	// first check ident name
+	identName := t.String()
+	_, ok := r.tableNames[identName]
```
Comment (qiuyesuifeng) [inline rset/rsets/join.go]: Seems can directly use `if r.tableNames[identName] { // do something }`?

Comment (siddontang) [inline rset/rsets/join.go]: I think use second ok for map is the common way for go.

**Comment by shenli** [comment]
LGTM

**Pull request merged**